tests/ct_commands.c

```diff
@@ -72,6 +72,135 @@ void test_mget(redisClusterContext *cc) {
     freeReplyObject(reply);
 }
 
+void test_hset_hget_hdel_hexists(redisClusterContext *cc) {
+    redisReply *reply;
+
+    // Prepare
+    reply = (redisReply *)redisClusterCommand(cc, "HDEL myhash field1");
+    CHECK_REPLY(cc, reply);
+    freeReplyObject(reply);
+    reply = (redisReply *)redisClusterCommand(cc, "HDEL myhash field2");
+    CHECK_REPLY(cc, reply);
+    freeReplyObject(reply);
+
+    // Set hash field
+    reply =
+        (redisReply *)redisClusterCommand(cc, "HSET myhash field1 hsetvalue");
+    CHECK_REPLY_INT(cc, reply, 1); // Set 1 field
+    freeReplyObject(reply);
+
+    // Set second hash field
+    reply =
+        (redisReply *)redisClusterCommand(cc, "HSET myhash field3 hsetvalue3");
+    CHECK_REPLY_INT(cc, reply, 1); // Set 1 field
+    freeReplyObject(reply);
+
+    // Get field value
+    reply = (redisReply *)redisClusterCommand(cc, "HGET myhash field1");
+    CHECK_REPLY_STR(cc, reply, "hsetvalue");
+    freeReplyObject(reply);
+
+    // Get field that is not present
+    reply = (redisReply *)redisClusterCommand(cc, "HGET myhash field2");
+    CHECK_REPLY_NIL(cc, reply);
+    freeReplyObject(reply);
+
+    // Delete a field
+    reply = (redisReply *)redisClusterCommand(cc, "HDEL myhash field1");
+    CHECK_REPLY_INT(cc, reply, 1); // Delete 1 field
+    freeReplyObject(reply);
+
+    // Delete a field that is not present
+    reply = (redisReply *)redisClusterCommand(cc, "HDEL myhash field2");
+    CHECK_REPLY_INT(cc, reply, 0); // Nothing to delete
+    freeReplyObject(reply);
+
+    // Check if field exists
+    reply = (redisReply *)redisClusterCommand(cc, "HEXISTS myhash field3");
+    CHECK_REPLY_INT(cc, reply, 1); // exists
+    freeReplyObject(reply);
+
+    // Delete multiple fields at once
+    reply = (redisReply *)redisClusterCommand(
+        cc, "HDEL myhash field1 field2 field3");
+    CHECK_REPLY_INT(cc, reply, 1); // field3 deleted
+    freeReplyObject(reply);
+
+    // Make sure field3 is deleted now
+    reply = (redisReply *)redisClusterCommand(cc, "HEXISTS myhash field3");
+    CHECK_REPLY_INT(cc, reply, 0); // no field
+    freeReplyObject(reply);
+
+    // As of Redis 4.0.0, HSET is variadic i.e. multiple field/value,
+    // but this is currently not supported.
+    reply = (redisReply *)redisClusterCommand(
+        cc, "HSET myhash field1 hsetvalue1 field2 hsetvalue2");
+    assert(reply == NULL);
+}
```

Comment:
Why is this not supported? The whole myhash is in the same cluster slot so the whole command can just be forwarded to one redis node, can't it?

Reply:
Exactly, bit wierd, but the parser dont handle this today. It put in the backlog, will be fixed in other PR